qlty-cli/src/commands/githooks/install.rs

```diff
@@ -0,0 +1,93 @@
+use crate::{Arguments, CommandError, CommandSuccess};
+use anyhow::{Context as _, Result};
+use clap::Args;
+use qlty_config::Workspace;
+use std::fs;
+use std::os::unix::fs::{symlink, PermissionsExt as _};
+use std::path::Path;
+
+#[derive(Args, Debug)]
+pub struct Install {}
+
+const QLTY_HOOKS_DIR: &str = ".qlty/hooks";
+
+const PRE_COMMIT_HOOK: &str = r#"#!/bin/sh
+qlty fmt --trigger pre-commit --index-file="$GIT_INDEX_FILE"
+"#;
+
+const PRE_PUSH_HOOK: &str = r#"#!/bin/sh
+qlty check --trigger pre-push --upstream-from-pre-push --no-formatters --skip-errored-plugins
+"#;
+
+impl Install {
+    pub fn execute(&self, _args: &Arguments) -> Result<CommandSuccess, CommandError> {
+        Workspace::require_initialized()?;
+
+        fs::create_dir_all(QLTY_HOOKS_DIR)?;
+
+        install_hook("pre-commit", PRE_COMMIT_HOOK)?;
+        install_hook("pre-push", PRE_PUSH_HOOK)?;
+
+        CommandSuccess::ok()
+    }
+}
+
+fn install_hook(hook_name: &str, contents: &str) -> Result<()> {
+    let script_filename = format!("{}.sh", hook_name);
+    let hook_script_path = Path::new(QLTY_HOOKS_DIR).join(script_filename.clone());
+    fs::write(&hook_script_path, contents).with_context(|| {
+        format!(
+            "Failed to write {} hook to {}",
+            hook_name,
+            hook_script_path.display()
+        )
+    })?;
+
+    let git_hooks_dir = Path::new(".git").join("hooks");
+    let symlink_path = git_hooks_dir.join(hook_name);
+
+    if symlink_path.exists() {
+        fs::remove_file(&symlink_path).with_context(|| {
+            format!(
+                "Failed to remove existing {} symlink at {}",
+                hook_name,
+                symlink_path.display()
+            )
+        })?;
+    }
+
+    let hook_relative_path = Path::new("..")
+        .join("..")
+        .join(".qlty")
+        .join("hooks")
+        .join(script_filename);
+
+    symlink(&hook_relative_path, &symlink_path).with_context(|| {
```

Comment:
Need to figure out what needs to be done to make this work on Windows

Reply:
This is how we do it elsewhere:

https://github.com/qltysh/qlty/blob/b5801ea27ca94c385b714d5ae93522da843863ed/qlty-check/src/executor/staging_area.rs#L288-L295

Reply:
What do you think about extracting this into a utility function which we could put in either a qlty-tips crate (which sits at the bottom of the dependency graph) or just drop into qlty-config initially which is a dependency of everything?